junipervpn/vpn.py

```diff
@@ -310,46 +311,45 @@ def stop(self):
 
 def main():
     parser = argparse.ArgumentParser(conflict_handler='resolve')
-    parser.add_argument('-h', '--host', type=str,
+    parser.add_argument('-h', '--host',
                         help='VPN host name')
-    parser.add_argument('-r', '--realm', type=str,
+    parser.add_argument('-r', '--realm',
                         help='VPN realm')
-    parser.add_argument('-u', '--username', type=str,
+    parser.add_argument('-u', '--username',
                         help='User name')
-    parser.add_argument('-p', '--pass_prefix', type=str,
+    parser.add_argument('-p', '--pass_prefix',
                         help="Secondary password prefix")
-    parser.add_argument('-o', '--oath', type=str,
+    parser.add_argument('-o', '--oath',
                         help='OATH key for two factor authentication (hex)')
-    parser.add_argument('-c', '--config', type=str,
+    parser.add_argument('-c', '--config',
                         help='Config file')
-    parser.add_argument('-s', '--stdin', type=str,
+    parser.add_argument('-s', '--stdin',
                         help="String to pass to action's stdin")
-    parser.add_argument('-d', '--device-id', type=str,
+    parser.add_argument('-d', '--device-id',
                         help="Hex device ID")
     parser.add_argument('-f', '--enable-funk', action='store_true',
+                        default=None,
                         help="Request funk message")
-    parser.add_argument('-H', '--hostname', type=str,
+    parser.add_argument('-H', '--hostname',
                         help="Hostname to pass with funk request")
-    parser.add_argument('-p', '--platform', type=str,
+    parser.add_argument('-p', '--platform',
                         help="Platform type to pass with funk request")
-    parser.add_argument('-a', '--hwaddr', type=str,
+    parser.add_argument('-a', '--hwaddr',
                         help="Comma separated list of hwaddrs to pass with funk request")
-    parser.add_argument('-C', '--certs', type=str,
+    parser.add_argument('-C', '--certs',
                         help="Comma separated list of pem formatted certificates for funk response")
-    parser.add_argument('-U', '--user-agent', type=str,
+    parser.add_argument('-U', '--user-agent',
                         help="User agent string")
     parser.add_argument('action', nargs=argparse.REMAINDER,
+                        default=[],
                         metavar='<action> [<args...>]',
                         help='External command')
 
     args = parser.parse_args()
-    args.__dict__['password'] = None
+    args.password = None
 
-    if len(args.action) and args.action[0] == '--':
-        args.action = args.action[1:]
-
-    if not len(args.action):
-        args.action = None
+    with contextlib.suppress(ValueError):
+        args.action.remove('--')
```

Comment:
Is this correct? `--` can only be the very first argument; we don't want to remove subsequent ones, because they could be part of the command to be executed.